auto_install/install.sh

```diff
@@ -500,6 +576,7 @@ preconfigurePackages(){
 		[[ $WIREGUARD_BUILTIN == 1 && ( $PLAT == 'Debian' || $PLAT == 'Raspbian' ) ]] ||
 		# If the module is not builtin, on Raspbian we know the headers package: raspberrypi-kernel-headers
 		[[ $PLAT == 'Raspbian' ]] ||
+		[ "${PLAT}" == 'Alpine' ] ||
```

Comment:
These checks are actually aimed to catch cases where the script can handle kernel source installation and WireGuard kernel module compiling correctly, which it doesn't for Alpine.

Is there a proper way to check whether the kernel comes with a pre-compile WireGuard module, so that `WIREGUARD_BUILTIN=1` can be set instead, like we do with APT? If it is in a container, it is indeed tricky. Probably a `modinfo wireguard 2> /dev/null` is then the only thing that can be done, which again depends on whether the container is actually able to read the hosts kernel modules that way. Otherwise it could be also blindly trusted that WireGuard is available, it should be on all non-ancient distro versions nowadays.

Reply:
There is a kernel module that will be loaded, but you need a compatible kernel version to do it (`linux-lts` or `linux-virt`, check it through `uname -mrs`)

https://wiki.alpinelinux.org/wiki/Configure_a_Wireguard_interface_(wg)

Reply:
Maybe I should change that piece of code with `[[ "${PLAT}" == 'Alpine' && "$(uname -mrs)" =~ ^Linux\s+[0-9\.\-]+-((lts)|(virt))\s+.*$ ]]`

Reply:
That would be an idea. But it doesn't cover containers (where Alpine is most used for) where the hosts kernel is relevant.

Would it be feasible to check for `grep -zaq '^container=' /proc/1/environ` and in case allow the WireGuard selection? Or can `modinfo wireguard`/`modprobe wireguard` succeed in Docker containers without manually mounting related paths into to container?

Reply:
`modprob` and `modinfo` doesn't work on Alpine Docker Image
```
# modprobe wireguard
modprobe: can't change directory to '/lib/modules': No such file or directory
modinfo wireguard
modinfo: can't open '/lib/modules/5.15.53-1-lts/modules.dep': No such file or directory
# ls -Ahl /lib/modules
ls: /lib/modules: No such file or directory
bash-5.1# ls -Ahl /lib/modules-load.d/
total 0
```

Reply:
`grep` doesn't work either
```
# grep -qzasEe '^container\=' /proc/1/environ; echo $?
1
```

Reply:
The condition `[[ "${PLAT}" == 'Alpine' && ! -f /.dockerenv && "$(uname -mrs)" =~ ^Linux\s+[0-9\.\-]+\-((lts)|(virt))\s+.*$ ]] ||` support the Alpine OS, while the condition `[[ "${PLAT}" == 'Alpine' && -f /.dockerenv && ... ]] ||` support the Alpine Docker Image
Only one point remains, how should the condition be for Alpine Docker Image ?

Reply:
@4s3ti @MichaIng and for this question ?

Reply:
@MichaIng do you have any idea ?